src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceDescriptor.cs

```diff
@@ -83,6 +106,14 @@ public static ServiceDescriptor Transient([NotNull] Type service, [NotNull] Type
             return Describe(service, implementationType, ServiceLifetime.Transient);
         }
 
+        public static ServiceDescriptor Transient<TService, TImplementation>(
+            [NotNull] Func<IServiceProvider, TImplementation> implementationFactory)
+            where TService : class
+            where TImplementation : class
```

Comment:
Seems reasonable to add generic constraints for TImplementation wherever possible. I'm not sure why these weren't included in the first place (ANI?) but I'll add them in.